capellambse/_diagram_cache.py

```diff
@@ -38,290 +39,185 @@
-        if self.image_format.lower() not in VALID_IMG_FMTS:
-            raise ValueError(
-                f"Invalid image format {self.image_format!r}, must be one of: "
-                + ", ".join(sorted(VALID_IMG_FMTS))
-            )
-        if self.force not in {"docker", "exe", None}:
-            raise ValueError(
-                "Invalid value for 'force', must be 'docker' or 'exe': "
-                f"{self.force}"
-            )
-        if self.force != "docker":
-            # use of Docker is not forced
-            if (exe := self._find_exe(self.capella_cli)) is not None:
-                # local exe has been identified
-                self._cli_cmd = [str(exe)]
-                self._cli_type = "exe"
-        if not self._cli_cmd and self.force != "exe":
-            # no local exe has been identified and Docker is allowed
-            if (exe := self._find_exe("docker")) is not None:
-                self._cli_cmd = [exe, "run", "--rm"]
-                self._cli_type = "docker"
-        if not self._cli_cmd:
-            if self.force == "exe":
-                raise FileNotFoundError(
-                    "Cannot find supplied 'capella_cli' executable: "
-                    f"{self.capella_cli}"
-                )
-            elif self.force == "docker":
-                raise FileNotFoundError(
-                    "Cannot find the executable 'docker' to run the "
-                    f"'capella_cli' image: {self.capella_cli}"
-                )
+        raise TypeError(
+            "Diagram cache updates are only supported for local paths"
+        )
+
+    diag_cache_dir = pathlib.Path(model._diagram_cache.path)
+    try:
+        shutil.rmtree(diag_cache_dir)
+    except FileNotFoundError:
+        pass
+    diag_cache_dir.mkdir(parents=True)
+
+    assert model.info.capella_version
+    capella = capella.replace("{VERSION}", model.info.capella_version)
+    native_args: dict[str, str] = {}
+    if force == "docker":
+        native_args["docker"] = capella
+    elif force == "exe":
+        native_args["exe"] = capella
+    else:
+        if pathlib.Path(capella).is_absolute():
+            native_args["exe"] = capella
+        elif pathlib.Path(capella).parent == pathlib.Path("."):
+            exe = shutil.which(capella)
+            if exe:
+                native_args["exe"] = exe
             else:
-                raise FileNotFoundError(
-                    "Cannot run 'capella_cli': Not an executable and "
-                    "'docker' was not found to run it as image: "
-                    f"{self.capella_cli}"
-                )
-        self._cli_common_flags = [
-            "-nosplash",
-            "-consolelog",
-            "-application",
-            "org.polarsys.capella.core.commandline.core",
+                raise ValueError(f"Not found in PATH: {capella}")
+        else:
+            native_args["docker"] = capella
+    assert "exe" in native_args or "docker" in native_args
```

Comment:
Could this be moved to a separate function, so it can be tested separately?
It looks like it's mixing different abstraction levels (model export and file path mangling).